### PR TITLE
operator manifests: Allow pinning opt-out

### PR DESCRIPTION
**Pull request by chmeliik**
* OSBS-8789

Users can configure whether to enable digest pinning and/or repo
replacement during operator manifest bundle builds in container.yaml.

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [ ] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by chmeliik** [comment]
Depends on https://github.com/containerbuildsystem/osbs-client/pull/962

**Comment by chmeliik** [comment]
Rebased

File: atomic_reactor/plugins/pre_pin_operator_digest.py [hunk 1]
```diff
@@ -171,24 +171,35 @@ def _get_replacement_pullspecs(self, pullspecs):
...
 
-            self.log.debug("Replacing namespace/repo")
-            repo_replaced = replacer.replace_repo(pinned)
+            if replace_repo:
+                self.log.debug("Replacing namespace/repo")
+                replaced = replacer.replace_repo(replaced)
 
             self.log.debug("Replacing registry")
-            registry_replaced = replacer.replace_registry(repo_replaced)
+            replaced = replacer.replace_registry(replaced)
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_pin_operator_digest.py:197]: I'm thinking if we should also add option for disabling registry replacements too

Comment (chmeliik) [reply atomic_reactor/plugins/pre_pin_operator_digest.py:197]: Would make sense. If all the replacements are disabled, should the plugin still check allowed_registries or just be skipped completely?

Comment (MartinBasti) [reply atomic_reactor/plugins/pre_pin_operator_digest.py:197]: it makes sense to skip it then, yup

**Comment by chmeliik** [comment]
Added option to disable registry replacements. If no replacements are enabled, the part where plugin computes replacements will be skipped. Technically, the plugin wouldn't have to run at all, but it can still serve as a sort of sanity check (is `manifests_dir` set properly, are all the CSV files and their pullspecs found etc.)

**Comment by rcerven** [comment]
code looks good, just test need fix

**Comment by chmeliik** [comment]
Restarted Travis build, the tests were dependent on the osbs-client PR (which is now merged)

**Comment by chmeliik** [comment]
> It would be nice to have the feature and tests in the same commit though (not a blocker for me).

Tests related to the atomic-reactor change are in the same commit. The 2nd commit is tests for new container.yaml entries added in an osbs-client PR, I'd rather keep it separate.

**Comment by chmeliik** [comment]
Release notes: pin_operator_digest now allows pullspec replacement opt-out via the `enable_{digest_pinning,repo_replacements,registry_replacements}` options in container.yaml

**Pull request merged**